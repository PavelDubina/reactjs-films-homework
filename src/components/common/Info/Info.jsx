/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import PropTypes from 'prop-types'
import { useSelector } from 'react-redux'
import className from 'classnames'
import { getGenres } from '../../../utils/getGenres'
import { Raiting } from '../Raiting/Raiting'
import styles from './Info.scss'

export const Info = ({ genreIds, title, vote, onClick, type }) => {
  const { genres } = useSelector((state) => state.genres)
  return (
    <div
      onClick={onClick}
      className={className({ [styles.info]: type === 'info', [styles.info_modal]: type === 'info-modal' })}
    >
      <div
        className={className({
          [styles.info__about]: type === 'info',
          [styles.info_modal__about]: type === 'info-modal',
        })}
      >
        <div
          className={className({
            [styles.info__about_title]: type === 'info',
            [styles.info_modal__about_title]: type === 'info-modal',
          })}
        >
          {title}
        </div>
        <Raiting vote={vote} />
      </div>
      <div
        className={className({
          [styles.info__genres]: type === 'info',
          [styles.info_modal__genres]: type === 'info-modal',
        })}
      >
        {getGenres(genreIds, genres)}
      </div>
    </div>
  )
}

Info.propTypes = {
  onClick: PropTypes.func,
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote: PropTypes.number.isRequired,
  genreIds: PropTypes.arrayOf(PropTypes.number).isRequired,
}
