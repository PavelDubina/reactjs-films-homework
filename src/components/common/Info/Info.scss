/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

.info {
  flex: 1.2;
  cursor: pointer;

  &__about {
    padding: 25px 20px 10px 20px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    &_title {
      font-size: 24px;
      padding-right: 10px;
      font-weight: bold;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    &_vote {
      font-weight: bold;
      color: rgb(24, 183, 236);
      padding: 1px 7px;
      border: 2px solid rgb(24, 183, 236);
      border-radius: 5px;
    }
  }

  &__genres {
    padding: 0 20px;
    color: rgb(24, 183, 236);
    font-weight: 800;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.info_modal {
  flex: 1;
  width: 100%;

  &__about {
    padding: 30px 0 5px 0;
    display: flex;
    justify-content: space-between;
    align-items: center;

    &_title {
      color: #fff;
      font-size: 24px;
      padding-right: 10px;
      font-weight: bold;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    &_vote {
      font-weight: bold;
      color: rgb(24, 183, 236);
      padding: 1px 7px;
      border: 2px solid rgb(24, 183, 236);
      border-radius: 5px;
    }
  }

  &__genres {
    color: rgb(24, 183, 236);
    font-weight: 800;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
