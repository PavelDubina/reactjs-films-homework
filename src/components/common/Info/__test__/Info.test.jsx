/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { genres } from '../../../../MockedData/mockedGenres'
import { Info } from '../Info'

const mockStore = configureStore([])

describe('Info', () => {
  let store
  beforeEach(() => {
    store = mockStore({
      genres: { genres },
    })
  })
  it('Should render and match snapshot', () => {
    const component = create(
      <Provider store={store}>
        <Info genreIds={[28, 12, 16]} title="title" vote={5} type="info" onClick={() => {}} />
      </Provider>,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
