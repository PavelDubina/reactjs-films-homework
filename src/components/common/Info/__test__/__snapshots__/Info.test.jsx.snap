// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Should render and match snapshot 1`] = `
<div
  className="info"
  onClick={[Function]}
>
  <div
    className="info__about"
  >
    <div
      className="info__about_title"
    >
      title
    </div>
    <div
      className="vote"
    >
      5
    </div>
  </div>
  <div
    className="info__genres"
  >
    Action, Adventure, Animation
  </div>
</div>
`;
